### PR TITLE
Remove projectroot dependency

### PR DESCRIPTION
**Pull request by vaibhavsagar**
Implement `getCabalProjectRootCurrent` as per #6.

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -90,9 +90,22 @@ getExecutables = do
...
+getCabalProjectRootCurrent = flip catchIOError (const (return Nothing)) $ do
+  cwd <- getCurrentDirectory
+  es  <- getDirectoryContents cwd
+  if any (isSuffixOf ".cabal") es
+    then return $ Just cwd
+    else do
+      let parent = takeDirectory cwd
+      if parent == cwd
+        then return Nothing
+        else withCurrentDirectory parent getCabalProjectRootCurrent
```
Comment (yamadapc) [inline src/Main.hs]: `withCurrentDictory` is the reason the build is failing; it isn't defined.

I'd rather have two functions: `getCabalProjectRootCurrent :: IO (Maybe FilePath)` and `getCabalProjectRoot :: FilePath -> IO (Maybe FilePath)`

Then the recursive call can just call the second version with `parent`, whereas the `getCabalProjectRootCurrent` version will just call the first with the current directory, without changing the environment or anything like that.

Comment (vaibhavsagar) [inline src/Main.hs]: I didn't realise `withCurrentDirectory` was a recent addition. I will reimplement it as you described.

File: src/Main.hs [hunk 2]
```diff
@@ -90,9 +90,24 @@ getExecutables = do
         [] -> error "No executables found"
         ds -> map fst ds
 
+getCabalProjectRootCurrent :: IO (Maybe FilePath)
+getCabalProjectRootCurrent = flip catchIOError (const (return Nothing)) $ do
```
Comment (vaibhavsagar) [inline src/Main.hs]: I'm not sure if `flip catchError (const (return Nothing))` is necessary here, but I thought I'd leave it in just to be sure.

**Comment by yamadapc** [comment]
Okay awesome! Thanks! This should work.

I'm going to try to fix the appveyor build and publish tomorrow.

**Pull request merged**